### PR TITLE
Updating configuration documentation.

### PR DESCRIPTION
**Pull request by rdegges**
Adding useful information, and a slightly more clear wording.

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -14,9 +14,11 @@ as **1 minute**!
...
 Environment Variables
 ---------------------
-Most Node.js applications expect your confidential information to be
-exposed by the environment (not hard-coded in the application).  You
-should export your Stormpath information by running this in the shell:
+
+It is a best practice to store confidential information in environment
+variables (*don't hard-code it into your application*). You should store your
+confidential Stormpath information in environment variables.  You can do this
+by running the following commands in the shell::
```
Comment (robertjd) [inline docs/configuration.rst]: Nneed to remove one of the colons at the end of this line, to get the code block to render propertly.

File: docs/configuration.rst [hunk 2]
```diff
@@ -24,10 +26,21 @@ should export your Stormpath information by running this in the shell:
     export STORMPATH_CLIENT_APIKEY_SECRET=YOUR-SECRET-HERE
     export STORMPATH_APPLICATION_HREF=YOUR-APP-HREF
 
+You might also want to check out
+`autoenv <https://github.com/kennethreitz/autoenv>`_, a project that makes
+working with environment variables simpler for Linux / Mac / BSD users.
+
```
Comment (robertjd) [inline docs/configuration.rst]: I love autoenv!  I think we should wrap this in a `.. tip::` block

File: docs/configuration.rst [hunk 3]
```diff
@@ -24,10 +26,21 @@ should export your Stormpath information by running this in the shell:
...
+    If you're on Windows, the command you need to use to set environment
+    variables is::
 
         set STORMPATH_CLIENT_APIKEY_ID=YOUR-ID-HERE
+        set STORMPATH_CLIENT_APIKEY_SECRET=YOUR-SECRET-HERE
+        set STORMPATH_APPLICATION_HREF=YOUR-APP-HREF
+
+The examples above show you the 3 mandatory settings you need to configure to
+make express-stormpath work.  These settings can be configured via environment
+variables, or in a number of other ways.
```
Comment (robertjd) [inline docs/configuration.rst]: Good clarification, I believe this paragraph should be moved below the code block that shows the export examples.

File: docs/configuration.rst (hunk 1 above)
Comment (rdegges) [inline docs/configuration.rst]: Thanks =)

File: docs/configuration.rst (hunk 2 above)
Comment (rdegges) [inline docs/configuration.rst]: Done!

File: docs/configuration.rst (hunk 3 above)
Comment (rdegges) [inline docs/configuration.rst]: Done.

**Comment by rdegges** [comment]
Ping @robertjd, made the updates, wanna merge?

**Comment by robertjd** [comment]
Thanks @rdegges !

**Pull request merged**